### PR TITLE
reinstate doc upload

### LINKED ISSUE #7577: live docs are out of date
#7407 migrated the doc building from CircleCI to GHA, but forgot the doc uploads. Thus, our last live doc build is from mid March.

cc @seemethere

### PR DESCRIPTION
**Pull request by pmeier**
Fixes #7577.

cc @seemethere

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -44,4 +45,58 @@ jobs:
...
+        
+        # On PRs we also want to upload the docs into our S3 bucket for preview.
+        if [[ ${{ github.event_name == 'pull_request' }} ]]; then
+          cp -r build/html/* "${RUNNER_DOCS_DIR}"
+        fi
+
+  upload:
+    # FIXME: this also needs to happen on created releases
+#    # FIXME: enable this before the PR is merged
+#    if: github.event_name == 'push' && github.event.push.base_ref == 'main'
```
Comment (pmeier) [inline .github/workflows/docs.yml:58]: As stated, I initially only want to run this on `main` and add the support for releases later on.

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -44,4 +45,58 @@ jobs:
...
+    needs: build
+    uses: pytorch/test-infra/.github/workflows/linux_job.yml@main
+    with:
+      repository: pytorch/vision
+      download-artifact: docs
+      ref: gh-pages
+      script: |
+        set -euo pipefail
+        
+        TARGET="main"
```
Comment (pmeier) [inline .github/workflows/docs.yml:68]: This is hardcoded now, but will have to be dynamic in the future.

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -44,4 +45,58 @@ jobs:
...
+        git config user.name "pytorchbot"
+        git commit -m "auto-generating sphinx docs" || true
+
+        # FIXME: this needs to come from ${{ secrets }}
+        export GITHUB_PYTORCHBOT_TOKEN=FOO
+        {
+          echo "machine github.com"
+          echo "login pytorchbot"
+          echo "password ${GITHUB_PYTORCHBOT_TOKEN}"
+        } > ~/.netrc
```
Comment (pmeier) [inline .github/workflows/docs.yml:92]: Previously, we had `GITHUB_PYTORCHBOT_TOKEN` available as environment variable. I already checked and it is not available through the GH secrets. On CIrcleCI we had

https://github.com/pytorch/vision/blob/82cf540b45fae6bc83cfff10db9041c3a1aeec5d/.circleci/config.yml#L919-L927

Not sure how to access the org-member context.

Comment (pmeier) [reply .github/workflows/docs.yml:92]: Maybe not, I honestly have not checked. Maybe we don't need it.

**Comment by pmeier** [comment]
I've tested this as best as I could on this PR:

- main branch / 21f2ce1c7752f07e2c6e793ab9be670a6d456d4d: https://github.com/pytorch/vision/actions/runs/5001497493/jobs/8960437744
- release tag / 9c26f0f651fcd8603b247052943cbb0d398a0edb: 
- RC tag / : 

I don't expect any major hiccups after merge. If there are some, I'm here to fix them.

**Review by NicolasHug — APPROVED** [review]
stamping, thanks Philip

**Pull request merged**